extension/dynamicconfig/model/pattern.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Contains common models for the dynamic config service. The corresponding
+// Proto() methods convert the model representation to a usable struct for
+// protobuf marshalling.
+
+package model
+
+import (
+	pb "github.com/vmingchen/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+)
+
+type Pattern struct {
```

Comment:
So we tell presence by checking whether they are empty? What if there is a "StartsWith" of an empty string to match all metrics?

Reply:
Great question; the zero-value for strings in Go is the empty string, so if no value for `StartsWith` is specified in the config, it will default to `""`. As a potential solution, I was thinking of having a special character to specify that we want to match everything, maybe `*`